apps/frontend/src/libs/modules/notification/notification.module.ts

```diff
@@ -0,0 +1,40 @@
+import { toast } from "react-toastify";
+
+import { type ValueOf } from "~/libs/types/types.js";
+
+import { NotificationType } from "./libs/enums/enums.js";
+import { type NotificationService } from "./libs/types/types.js";
+
+type Options = {
+	type: ValueOf<typeof NotificationType>;
+};
+
+const DEFAULT_MESSAGE = "Unexpected error";
+
+class Notification implements NotificationService {
```

Comment:
```suggestion
class Notification {
```
you do not need an additional type for implementation